packages/delivery-electron/delivery.js

```diff
@@ -6,28 +6,46 @@ const NetworkStatus = require('@bugsnag/electron-network-status')
 
 const delivery = (client, filestore, net, app) => {
   const send = (opts, body, cb) => {
-    const errorHandler = err => {
-      err.isRetryable = true
-      cb(err)
+    // an error can still happen on both the request and response even after a response is received,
+    // so we store a reference to the response error to ensure this is only handled once
+    let responseError
+
+    const requestErrorHandler = requestError => {
+      if (responseError) return
+
+      requestError.isRetryable = true
```

Comment:
**thought:** I didn't have a good overall understanding of bugsnag codebase, but I'd suggest (a very aggressive approach) `isRetryable = false` for any network errors.

In terms of Desktop applications, we don't know (or even cannot predict) how users setup their machines locally. Things like firewalls, enterprise proxies, dns blockers (add blockers) or unexpected stuff would make the request to Bugsnag fail. We should be really careful when it comes to retryable request, to avoid any possible infinite loop

Reply:
Thanks for the comment @trungutt, and for raising this issue initially!

The `isRetryable` flag is there to determine whether events that fail to send should be persisted and retried later or whether they should be discarded completely.

The potential for an infinite request loop isn't related to whether or not an error is marked as retryable - it happens because the error events emitted by Electron net on the request object and response stream after a response is received were not being handled, so these would then fall through to the uncaught exception handler which would then try and notify that error as a separate, duplicate error regardless of whether or not the initial failure was retryable. Now that these error events are being handled properly, this is no longer the case.